### PR TITLE
Add test for upper/lower/titlecase and fix call

### PR DESCRIPTION
**Pull request by kshyatt**
Looks like there's no `annotations` function defined so I added a direct reference to the field. Can always add a getter function if preferred.

### CODE REVIEW

File: test/strings/annotated.jl [hunk 1]
```diff
@@ -64,6 +64,9 @@ end
 @testset "AnnotatedChar" begin
     chr = Base.AnnotatedChar('c')
     @test chr == Base.AnnotatedChar(chr.char, Pair{Symbol, Any}[])
+    @test uppercase(chr) == Base.AnnotatedChar('C')
+    @test titlecase(chr) == Base.AnnotatedChar('C')
+    @test lowercase(Base.AnnotatedChar('C')) == chr
```
Comment (giordano) [inline test/strings/annotated.jl:69]:
```suggestion
    @test lowercase(Base.AnnotatedChar('C')) == chr
```

**Comment by kshyatt** [comment]
Wooo tests passed. Good to merge?

**Comment by giordano** [comment]
> Looks like there's no annotations function defined 

What happened to
https://github.com/JuliaLang/julia/blob/faa6095e983fed777007274f4efb9911154b1f3d/base/strings/annotated.jl#L389
? Why is that not available?

**Comment by kshyatt** [comment]
Exactly. We can also extend it to the char type if you like

**Comment by giordano** [comment]
@tecosaur did you mean to have this method?

**Review by LilithHafner — APPROVED** [review]

**Comment by tecosaur** [comment]
> did you mean to have this method?

Yea, we should have it. I think it just slipped my mind at the time.

**Comment by kshyatt** [comment]
@tecosaur you want me to add it as part of this PR?

**Comment by tecosaur** [comment]
That would be great, thanks for giving this a look.

**Comment by kshyatt** [comment]
actually there *is* an `annotations` method defined for `AnnotatedChar`. I'll try to figure out why it's not working here.

**Comment by kshyatt** [comment]
missing import of `annotations` into `base/strings/unicode.jl` now fixed

**Pull request merged**